### PR TITLE
Update the pull request template

### PR DESCRIPTION
**Pull request by acdaigneault**
### Description of the problem

It was easy to forget a few tasks related to the pull request (indent, tests, change log, etc...). 
This PR adds multiple templates depending on the category (new feature, fix, enhancement, doc or example) and adds a checklist.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,47 +9,50 @@ is welcome to do so.
...
+  * If the PR adds a new feature:\
+    The feature should be documented, tested (with unit tests and/or application tests) 
+    and if the feature adds or alters parameters in the input file, the appropriate section 
+    of the documentation should be updated accordingly.
+  * If the pull request corrects a bug:\
+    the source of the bug should be explained and the way it was identified should be briefly described. 
+    A unit test or application test that reproduces the bug should be added.
+
+See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.
+
```
Comment (AmishgaAlphonius) [inline CONTRIBUTING.md:39]: Unrelated:
This is maybe not for this PR and should be added to the "Miscellaneous" subsection of the PR, but a check should be done to make sure that all information here correspond to what's on the online documentation and vice-versa.
For example, the 60 char rule is not mentioned in the online doc and in this file the `prmindent` is not mentioned. However, since you link the doc here, you don't need to put all the info that's there here, but there is no link from the doc to here so the other way is the still a problem.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,31 +1,39 @@
...
+- [ ] All in-code documentation related to this PR is updated (Doxygen format)
+- [ ] Lethe documentation is updated
+- [ ] If possible, new feature has a unit test or application test
+- [ ] Changelog (CHANGELOG.md) is updated
+- [ ] Code is indented with indent-all and .prm files with prm-indent
 
-- Anything that you want to add up that didn't fit in any other category can be written here
+Pull request related list:
+- [ ] Labels have been applied
+- [ ] There are at least 2 reviewers (or 1 if small feature) excluding the merger
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE.md:38]:
```suggestion
- [ ] There are at least 3 reviewers (or 2 if small feature) excluding the merger
```
Isn't it 2 except Bruno so 3, or do we keep 2 and accept that implicitly it means that Bruno does not count for one of them?
image

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -9,47 +9,50 @@ is welcome to do so.
...
 
 # Good practices
 
 - Before making a pull request, the branch should be rebased on master to ensure
```
Comment (AmishgaAlphonius) [inline CONTRIBUTING.md:43]:
```suggestion
- Before making a pull request, the branch should be rebased onto master to ensure
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -9,47 +9,50 @@ is welcome to do so.
...
-a linear history and to make merging as easy as possible. It is the responsability
+a linear history and to make merging as easy as possible. It is the responsibility
 of the branch owner to ensure that the pull request goes as smoothly as possible.
-Pushing to github after a rebase requires a force push. You can accomplish
+Pushing to Github after a rebase requires a force push. You can accomplish
 this by using `git push --force-with-lease`
 - All functions should be documented in the `.h`. All functions should contain
 a `@brief` that describes the function and a `@param` for each argument.
```
Comment (AmishgaAlphonius) [inline CONTRIBUTING.md:49]: Unrelated:
```suggestion
- All functions should be documented in the `.h`. All functions should contain
a `@brief` that describes the function, a `@param` for each argument, and a `@return` if anything is returned by the function.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -95,11 +98,11 @@ determined by the headers included in the source files and headers of
 those libraries and executables) in the `CMakeLists.txt` files'
 `TARGET_LINK_LIBRARIES` calls, and omit any transitive dependencies.
 
-For example, the `dem_2d` application
+For example, the `lethe-particles` application
 [links to `lethe-core` and `lethe-dem`](applications/dem_2d/CMakeLists.txt)
```
Comment (AmishgaAlphonius) [inline CONTRIBUTING.md:102]:
```suggestion
[links to `lethe-core` and `lethe-dem`](applications/lethe-particles/CMakeLists.txt)
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -95,11 +98,11 @@ determined by the headers included in the source files and headers of
...
 because it
 [includes core and DEM headers](applications/dem_2d/dem_2d.cc),
-but the `gd_navier_stokes_2d` application
+but the `lethe-fluid` application
 [links only to `lethe-solvers`](applications/gd_navier_stokes_2d/CMakeLists.txt)
```
Comment (AmishgaAlphonius) [inline CONTRIBUTING.md:106]:
```suggestion
[links only to `lethe-solvers`](applications/lethe-fluid/CMakeLists.txt)
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -95,11 +98,11 @@ determined by the headers included in the source files and headers of
 those libraries and executables) in the `CMakeLists.txt` files'
 `TARGET_LINK_LIBRARIES` calls, and omit any transitive dependencies.
 
-For example, the `dem_2d` application
+For example, the `lethe-particles` application
 [links to `lethe-core` and `lethe-dem`](applications/dem_2d/CMakeLists.txt)
 because it
 [includes core and DEM headers](applications/dem_2d/dem_2d.cc),
```
Comment (AmishgaAlphonius) [inline CONTRIBUTING.md:104]:
```suggestion
[includes core and DEM headers](applications/lethe-particles/dem.cc),
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -95,11 +98,11 @@ determined by the headers included in the source files and headers of
...
-but the `gd_navier_stokes_2d` application
+but the `lethe-fluid` application
 [links only to `lethe-solvers`](applications/gd_navier_stokes_2d/CMakeLists.txt)
 because it
 [includes only solvers headers](applications/gd_navier_stokes_2d/gd_navier_stokes_2d.cc),
```
Comment (AmishgaAlphonius) [inline CONTRIBUTING.md:108]:
```suggestion
[includes only solvers headers](applications/lethe-fluid/gls_navier_stokes.cc),
```

File: doc/source/contributing.rst [hunk 9]
```diff
@@ -34,6 +35,14 @@ Reviewers Responsibility
 * Verify that the pull request meets the requirement described above regarding testing and the format of the pull request.
 * Give the review in a timely manner.
 
+Description
+-----------
+
+Without explicit request to the merger, all commits will be squashed when merging. The pull request description will be the only remaining description of the changes left in the code history.
```
Comment (AmishgaAlphonius) [inline doc/source/contributing.rst:41]:
```suggestion
Without explicit request to the person responsible for merging the PR, all commits will be squashed when merging. The pull request description will be the only remaining description of the changes left in the code history.
```

**Review by AmishgaAlphonius — COMMENTED** [review]
It's a great idea to add this checklist and since it will be removed during merger it won't make the commit message unnecessarily longer. Nice work!! :)
Most of my comments and suggestions are not related to this PR's main objective, but are concerned by the files touched here.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE.md:38]: 2 excluding one means 3 for me, am I wrong? 
By merger (the person that does the merge in Master) I mean Bruno or the person when Bruno is in vacations

File: CONTRIBUTING.md (hunk 1 above)
Comment (acdaigneault) [reply CONTRIBUTING.md:39]: Actually, if the info is somewhere, it should not be somewhere else also. Meaning, we should refer to the doc only. 
Let see what Bruno has to say about that

File: CONTRIBUTING.md (hunk 4 above)
Comment (acdaigneault) [reply CONTRIBUTING.md:49]: I did not know we needed the @return actually. I guess our "official" Doxygen rules should be all listed in the contributing doc and not here

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -1,31 +1,39 @@
...
 
-# Comments
+# Checklist (will be removed when merged)
+See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.
+
+Code related list:
+- [ ] All in-code documentation related to this PR is updated (Doxygen format)
+- [ ] Lethe documentation is updated
+- [ ] If possible, new feature has a unit test or application test
+- [ ] Changelog (CHANGELOG.md) is updated
```
[the remark this answers is not in the document]
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE.md:33]: I think, but not sure, it's only related to code development. Like, to be aware of modifications, fixes etc

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -1,31 +1,39 @@
...
+- [ ] Code is indented with indent-all and .prm files with prm-indent
```
[the remark this answers is not in the document]
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE.md:34]: Ohh yes nice catch

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -9,47 +9,50 @@ is welcome to do so.
 This document aims at establishing guidelines for contributions within
 Lethe.
 
-Lethe has a wiki which is openly accessible on : https://github.com/chaos-polymtl/lethe/wiki
+Lethe has a wiki which is openly accessible on : https://chaos-polymtl.github.io/lethe/index.html
```
[the remark this answers is not in the document]
Comment (acdaigneault) [reply CONTRIBUTING.md:12]: I agree it does not anymore

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (AmishgaAlphonius) [reply .github/PULL_REQUEST_TEMPLATE.md:39]: Ohh, you're right that's perfectly fine, I misread the last part sorry 😅

However, I think merger, should be replaced by "the person that merges the PR" (or some other alternative), because "merger" in the way you use it here is not really defined that way in the English language.

File: CONTRIBUTING.md (hunk 1 above)
Comment (AmishgaAlphonius) [reply CONTRIBUTING.md:38]: I remember that there was an issue regarding the fact that when the documentation was down (during the time doxygen was added) it was not possible to know how to locally compile from the documentation online. Other than that (how to locally compile documentation), I think you're right and we could leave the rest online.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (blaisb) [reply .github/PULL_REQUEST_TEMPLATE.md:34]: Changelog should be related to the software itself. Either new features, bugs that are fixed, parameter names that are changed and so on and so forth

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 13]
```diff
@@ -1,31 +1,40 @@
-# Description of the problem
+# Description of the issue
```
Comment (blaisb) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: I am unsure if the word issue is the right thing here right. Because sometimes a pull request does not fix an issue.
Maybe the phrasing "Description of the content of the pull request"

It could be possible to merge the two paragraph.
Like description of the content and then we explain if this is a fix what is the issue and how it is fixed. If it's a feature explain what is the feature and what it accomplishes

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 14]
```diff
@@ -1,31 +1,40 @@
...
-# How Has This Been Tested?
+# How has this been tested?
 
-- List and describe the test(s) for the added code (if there is no test for the pull request, explain why (i.e. the test already exists but was changed))
-
-- [ ] Test A
-- [ ] Test B
+- List and describe the new test(s) (i.e. what feature/parameter does it test)
+- Explain the changes and/or impacts on current tests
+- If not related to tests, how did you ensure that the solution works?
```
Comment (blaisb) [inline .github/PULL_REQUEST_TEMPLATE.md:15]:
```suggestion
- If not related to tests, how did you ensure that the solution works and how will you ensure that it will continue to work in the future?
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 15]
```diff
@@ -1,31 +1,40 @@
...
-# Comments
-
-- Anything that you want to add up that didn't fit in any other category can be written here
+# Checklist (will be removed when merged)
+See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.
+
+Code related list:
+- [ ] All in-code documentation related to this PR is updated (Doxygen format)
+- [ ] Lethe documentation is updated
+- [ ] If possible, new feature has a unit test or application test
```
Comment (blaisb) [inline .github/PULL_REQUEST_TEMPLATE.md:32]:
```suggestion
- [ ] If possible, new feature has a unit test (preferred) or application test
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (blaisb) [reply .github/PULL_REQUEST_TEMPLATE.md:39]: Let's keep 2 reviewers for big features and 1 for small feature. Let'S just exclude me in this list. This way I can help in the end.

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -1,55 +1,57 @@
 # Lethe
 
-Lethe is an open source and we try to follow the deal.II mentality of having it
-developed in an open accessible environment. Consequently, all scientific
+Lethe is an open source software, and we try to follow the deal.II mentality of having it
+developed in an accessible environment. Consequently, all scientific
 developments made within Lethe are accessible to everyone even as they are
```
Comment (blaisb) [inline CONTRIBUTING.md:5]:
```suggestion
developments made within Lethe are available to everyone even as they are
```

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -1,55 +1,57 @@
...
 currently being written. Anybody that wishes to use and contribute to a feature
 is welcome to do so.
```
Comment (blaisb) [inline CONTRIBUTING.md:7]:
```suggestion
is welcomed to do so.
```

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -95,14 +97,14 @@ determined by the headers included in the source files and headers of
...
 because it
-[includes core and DEM headers](applications/dem_2d/dem_2d.cc),
-but the `gd_navier_stokes_2d` application
-[links only to `lethe-solvers`](applications/gd_navier_stokes_2d/CMakeLists.txt)
+[includes core and DEM headers](applications/lethe-particles/dem_2d.cc),
+but the `lethe-fluid` application
+[links only to `lethe-solvers`](applications/lethe-fluid/CMakeLists.txt)
 because it
-[includes only solvers headers](applications/gd_navier_stokes_2d/gd_navier_stokes_2d.cc),
+[includes only solvers headers](applications/lethe-fluid/gd_navier_stokes_2d.cc),
```
Comment (blaisb) [inline CONTRIBUTING.md:107]: We should put the name of the executable as they are now here. dem_2d and gd_navier_stokes_2d do not exist anymore. :)

**Review by blaisb — CHANGES_REQUESTED** [review]
A few changes, thank you for the very good work!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 13 above)
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: I kinda like the distinction between problem/solution. When I reread the commit message while debugging it is easier to target what was done since I don't really need to know why it was done. I really hate when people write "this is fixed" in the solution tho :P 
I agree with "content of PR"

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 14 above)
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: I understand why it is good to know, but I think this is something we can hardly anwser. I feel nobody will keep that one. Do you have any example?

**Comment by acdaigneault** [comment]
Guys I found out how to create multiple template. So I will do individual templates for each categories and continue to apply your suggestions!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 14 above)
Comment (blaisb) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: Yeah you are right. It's wishful thinking more than anything else. Maybe this is not necessary in the end.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 13 above)
Comment (blaisb) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Yup agreed. Maybe this is something I can fix though. I can remind people that  PR messages should be more descriptive. I am also guilty of this to be honest. I will be more careful from now on.

**Comment by acdaigneault** [comment]
I added the new templates. 
So the default template has links that append the URL to the chosen PR template as: 
image
image

And by instance, the template markdown will look like this:
image

And the result will be:
image

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md [hunk 19]
```diff
@@ -0,0 +1,48 @@
+<!-- Please, fill in the description as completely as possible.-->
+
+### Description
+
+<!-- Explain the issue with the bug (what part of the code, what are the side effects of the bug)
+       How did the bug was found, do you know what commit introduced the bug? -->
```
Comment (acdaigneault) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md:6]: The extra spaces are made on purpose since the markdown code on Github is not monospace and I wanted it flushed with the first line

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+       Are there changes and/or impacts on current tests, why?
+       How did you ensure that the solution works? -->
+
+### Documentation
+
+<!-- Does this fix modify or have new simulation parameters? If so, describe them. -->
+
+### Miscellaneous (will be removed when merged)
+
+<!-- Anything that you would like to add that does not fit into another category
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md:26]:
```suggestion
<!-- Anything that you would like to add that does not fit into any of the categories above.
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+       Examples:
+         Future changes or features that will be added in subsequent pull requests
+         Any comments or highlights for the reviewers -->
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md:29]:
```suggestion
         Future changes or features that will be added in subsequent pull requests.
         Any comments or highlights for the reviewers. -->
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.
+*If applicable
+
+Code related list:
+- [ ] All in-code documentation related to this PR is updated (Doxygen format)*
+- [ ] Lethe documentation is updated*
+- [ ] Fix has unit test(s) (preferred) or application test(s)
+- [ ] The branch has been rebased onto master
+- [ ] Changelog (CHANGELOG.md) is updated
+- [ ] Code is indented with indent-all and .prm files (examples and tests) with prm-indent
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md:41]: (same for other files)

```suggestion
- [ ] All in-code documentation related to this PR has been updated (Doxygen format)*
- [ ] Lethe documentation has been updated*
- [ ] Unit test(s) (preferred) or application test(s) have been added to test the fix
- [ ] The branch has been rebased onto master
- [ ] Changelog (CHANGELOG.md) has been updated
- [ ] Code has been indented with indent-all and .prm files (examples and tests) with prm-indent
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+
+Pull request related list:
+- [ ] Labels have been applied
+- [ ] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
+- [ ] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
+- [ ] If temporary fix, an issue is opened
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md:47]:
```suggestion
- [ ] If the fix is temporary, an issue is opened
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_documentation.md [hunk 24]
```diff
@@ -0,0 +1,29 @@
+<!-- Please, fill in the description as completely as possible.-->
+
+### Description
+
+<!-- Explain the content of this documentation
+       It is related to Lethe documentation (simulation parameters or theory guide) or in-code documentation? -->
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_documentation.md:6]:
```suggestion
       Is it related to Lethe documentation (simulation parameters or theory guide) or in-code documentation? -->
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_example.md [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+         Any comments or highlights for the reviewers -->
+
+### Checklist (will be removed when merged)
+See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.
+
+Code related list:
+- [ ] Lethe documentation is updated
+- [ ] The branch has been rebased onto master
+- [ ] Code is indented with indent-all and .prm files with prm-indent
+
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_example.md:23]:
```suggestion
- [ ] Links have been added to parent .rst files.

```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_new-feature.md [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+
+<!-- How has this been tested?
+       What are the new test(s) and what feature/parameter does it test?
+       Are there changes and/or impacts on current tests, why?
+       How did you ensure that the solution works?
+       How will you ensure that it will continue to work in the future? -->
+
+### Documentation
+
+<!-- Does this new feature have new simulation parameters? If so, describe them. -->
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_new-feature.md:19]:
```suggestion
<!-- Does this new feature introduce new simulation parameters? If so, describe them. -->
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_new-feature.md [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+       What are the motivations? 
+       How is it integrated to the current code? -->
+
+### Testing
+
+<!-- How has this been tested?
+       What are the new test(s) and what feature/parameter does it test?
+       Are there changes and/or impacts on current tests, why?
+       How did you ensure that the solution works?
+       How will you ensure that it will continue to work in the future? -->
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_new-feature.md:15]:
```suggestion
<!-- How has this been tested?
       What are the new test(s) and what feature(s)/parameter(s) does it test?
       Are there changes and/or impacts on current tests, why?
       How did you ensure that the solution works?
       How will you ensure that it will continue to work in the future? -->
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_refactoring.md [hunk 28]
```diff
@@ -0,0 +1,40 @@
+<!-- Please, fill in the description as completely as possible.-->
+
+### Description
+
+<!-- Explain the aim of this refactoring
+       What are the motivations? 
+       How is it integrated to the current code? -->
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_refactoring.md:7]:
```suggestion
<!-- Explain the aim of this refactoring.
       What are the motivations? 
       How is it integrated to the current code? -->
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md [hunk 29]
```diff
@@ -0,0 +1,48 @@
...
+### Miscellaneous (will be removed when merged)
+
+<!-- Anything that you would like to add that does not fit into another category
+       Examples:
+         Future changes or features that will be added in subsequent pull requests
+         Any comments or highlights for the reviewers -->
+
+### Checklist (will be removed when merged)
+See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.
+*If applicable
```
Comment (AmishgaAlphonius) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md:33]: Maybe move this line at the bottom? It confused me at first, but maybe it's just me.
(same for other files with it)

File: CONTRIBUTING.md [hunk 30]
```diff
@@ -1,55 +1,57 @@
...
 currently being written. Anybody that wishes to use and contribute to a feature
-is welcome to do so.
+is welcomed to do so.
 
-This document aims at establishing guidelines for contributions within
-Lethe.
+This document aims at establishing guidelines for contributions within Lethe.
 
-Lethe has a wiki which is openly accessible on : https://github.com/chaos-polymtl/lethe/wiki
+Lethe has its own documentation (user guide and developer guide) which are openly accessible on: https://chaos-polymtl.github.io/lethe/index.html
```
Comment (AmishgaAlphonius) [inline CONTRIBUTING.md:11]:
```suggestion
Lethe has its own documentation (user guide and developer guide) which are available on: https://chaos-polymtl.github.io/lethe/index.html
```

**Review by AmishgaAlphonius — COMMENTED** [review]
Last few comments/suggestions :)

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md [hunk 31]
```diff
@@ -0,0 +1,48 @@
...
+<!-- How has this been tested?
+       What are the new test(s) that reproduce the bug?
+       Are there changes and/or impacts on current tests, why?
+       How did you ensure that the solution works? -->
+
+### Documentation
+
+<!-- Does this fix modify or have new simulation parameters? If so, describe them. -->
+
+### Miscellaneous (will be removed when merged)
```
[the remark this answers is not in the document]
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md:24]: No this is not a mistake. 
Anything that do not fit in other category in more related to details or communication to reviewers or a trace of something that will still be in kept in the PR archive (nothing is changed, neither the checklist), but are not relevant enough to be in the final commit message.
This is to simply the number of sections.
What do you think?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 32]
```diff
@@ -1,31 +1,7 @@
...
-- If applicable, describe future changes or features that will be added in subsequent pull requests
-
-# Comments
-
-- Anything that you want to add up that didn't fit in any other category can be written here
+* :bulb: [New feature](?expand=1&template=pull_request_template_new-feature.md) 
+* :wrench: [Bug fix](?expand=1&template=pull_request_template_bug-fix.md)
+* :books: [Documentation](?expand=1&template=pull_request_template_documentation.md)
+* :sparkles: [Refactoring](?expand=1&template=pull_request_template_refactoring.md) 
+* :crystal_ball: [Example](?expand=1&template=pull_request_template_example.md)
```
[the remark this answers is not in the document]
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Yeah the cristal ball was not an easy choice nor the best haha

File: CONTRIBUTING.md (hunk 30 above)
Comment (acdaigneault) [reply CONTRIBUTING.md:11]: Bruno suggested to change accessible to available, will let you fight on that one 👀

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_example.md (hunk 25 above)
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template_example.md:23]: Nice catch

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md (hunk 22 above)
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md:41]: I must say that I don't fully agree with those changes, but let's discuss here

For me, giving the statement that something have been done means that an action related to modification has been perform. But, what I wanted the checklist to focus on, is the state of the thing and not the process.

For instance with the documentation:
Lethe documentation is updated vs. Lethe documentation has been updated
For me, #1 means the author made sure that the documentation is updated related to its work, maybe the person had to modify documentation or not. 
#2 means that the author had to modify the documentation.

Maybe it is a language interpertation thing, but I think "is" is better
However, it highlights the irrevelancy of the *if application, that I will remove.

What do you think about it :) ?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md (hunk 31 above)
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md:24]: You're right it was just for the future works, that I was thinking that it still might be nice to have them in the message, but maybe it's not necessary.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md (hunk 29 above)
Comment (acdaigneault) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template_bug-fix.md]: It's removed now!

**Review by AmishgaAlphonius — APPROVED** [review]

**Review by blaisb — APPROVED** [review]
I'm very happy with these changes. I think we can go with that and see organically how things evolve from there :)

**Comment by acdaigneault** [comment]
Ok, tell me when it is merged, I will do a post about it in #Lethe on Slack
